launcher/src/main/java/org/apache/spark/launcher/SparkSubmitCommandBuilder.java

```diff
@@ -154,9 +165,17 @@
 
   List<String> buildSparkSubmitArgs() {
     List<String> args = new ArrayList<>();
-    SparkSubmitOptionParser parser = new SparkSubmitOptionParser();
+    OptionParser parser = new OptionParser(false);
```

Comment:
whats the reason for allowing unknown args here?  Is it so an old launcher can start a newer spark, which may accept more args?  A comment would be helpful